scripts/gen_app_partitions.py

```diff
@@ -25,7 +32,7 @@
 """
 
 library_data_template = """
-		*{0}:*(.data .data.*)
+		*{0}:*(SORT_BY_ALIGNMENT(.data) SORT_BY_ALIGNMENT(.data.*))
```

Comment:
strictly speaking, if I understand correctly this change is orthogonal to this patch and is more related to #14183 can you make this change there instead?